src/core/flow.ts

```diff
@@ -23,6 +23,7 @@ import type {
   ActionShape,
   State,
   Ctx,
+  FnContext,
```

Comment:
I do agree with the point that it probably makes sense to call the aggregate (G + ctx + ...) as `Context`, which begs the question about what to do to `ctx`.

Do you think `ctx: GameContext` is a suitable name combination?   I'm less convinced by `flow`.

Reply:
I’ll try to think further on what it might be named. I suppose I’m trying to avoid `context.ctx` as it seems rather confusing and vague to have a context inside a context. Some other options to brainstorm from (but something terser would be better): `playState`, `flowState`, `engineState`. I personally like how `flow` combines with the various members, e.g. `flow.currentPlayer`, `flow.playOrder`, `flow.phase` etc. while not being too verbose, but perhaps there’s something better.

That said, `G` + `ctx` is a pretty core concept for current users, so I could also definitely understand keeping it as is if we can’t find anything better. (Although previously a move could also have been `(G, foo, ...args) => {}` so `ctx` was only the conventional name for that second argument.)

Reply:
I dont think this is a major problem, because the Context in this case is only used internally. Most of our docs were updated to use the object destructuring assignments, so it doesnt look confusing for users.